AEPCore/Sources/rules/LaunchRulesEngine.swift

```diff
@@ -330,6 +336,123 @@ public class LaunchRulesEngine {
         dict[LaunchRulesEngine.CONSEQUENCE_EVENT_DATA_KEY_TYPE] = consequence.type
         return parentEvent.createChainedEvent(name: LaunchRulesEngine.CONSEQUENCE_EVENT_NAME, type: EventType.rulesEngine, source: EventSource.responseContent, data: [LaunchRulesEngine.CONSEQUENCE_EVENT_DATA_KEY_CONSEQUENCE: dict])
     }
+
+    /// Process a schema consequence event. Handles different schema types including event history operations.
+    /// - Parameters:
+    ///   - consequence: the RuleConsequence which contains the schema details
+    ///   - processedEvent: the event that triggered the rule
+    ///   - data: a `Traversable` object used for token resolution
+    private func processSchemaConsequence(consequence: RuleConsequence, processedEvent: Event, data: Traversable) {
+        guard let schema = consequence.schema else {
+            Log.error(label: LOG_TAG, "(\(self.name)) : Unable to process a Schema Consequence, 'schema' is missing from 'details'")
```

Comment:
nit: As per the [definition](https://wiki.corp.adobe.com/display/ADMSMobile/Schema-based+consequence+details#Schemabasedconsequencedetails-ConsequenceType) of `schema` consequence type, the id and data are also mandatory fields. Wondering if we should have that check here and fast fail